### DISCUSSION TITLE
Anybody know why web app required react-native?

### DISCUSSION THREAD

**Discussion by victorpavlenko** [wevm/wagmi#2274] [Q&A] [answered]
```
WARNING in ./node_modules/@walletconnect/universal-provider/dist/index.es.js 3968:105-144
Module not found: Error: Can't resolve 'react-native' in '/Users/sasa/web-app/node_modules/@walletconnect/universal-provider/dist'
```

Should I add it into dep?

**Comment by jxom** [comment, marked as answer]
This is from a WalletConnect dependency. They are working to solve the issue.

**Reply by allenchuang** [reply]
following 👀

**Reply by ghost** [reply]
Me too following... 👀

**Reply by artschekoff** [reply]
im in guys

**Reply by federicolopezeikilis** [reply]
following!

**Reply by LinusMelb** [reply]
following

**Reply by cuongpham2107** [reply]
following

**Comment by cryptod00c** [comment]
this is incredibly annoying. can I not just opt out of it on the wagmi side?

Also for the wagmi team - your migration docs are incredibly well written - and this error will continue to come up for 99% of the people migrating making them frustrated with the migration and not @ walletconnect where the disappointment should be angled.

what a massive oversight for the walletconnect team.

**Reply by jxom** [reply]
The way to opt-out would be to not use WalletConnectConnector. 😅

Note that this is just a warning from the bundler that the module doesn’t exist, the `react-native` dependency isn’t actually executed in wagmi.

**Reply by cryptod00c** [reply]
It's an error in connectkit - that prevents the build from completing. They have a PR open to merge in WAGMI v1.

Very excited about this upgrade and the move to Viem

**Reply by The18thWarrior** [reply]
Is there a workaround to bypass the error for a production build?

**Reply by cryptod00c** [reply]
@The18thWarrior add this to your `next.config.js` or webpack settings...

```js
webpack: (config, { isServer, webpack }) => {
    if (!isServer) {
      config.resolve.fallback = {
        fs: false,
        net: false,
        tls: false,
        encoding: false,
        // crypto: require.resolve('crypto-browserify'),
      };
    }

    config.resolve.alias = {
      ...config.resolve.alias,
      'react-native$': 'react-native-web',
    };
    config.resolve.extensions = [
      '.web.js',
      '.web.ts',
      '.web.tsx',
      ...config.resolve.extensions,
    ];

    return config;
  },
  ```

**Reply by The18thWarrior** [reply]
Cool, I'll try that out.

**Reply by The18thWarrior** [reply]
Still getting the same error ☹️

**Reply by cryptod00c** [reply]
what's the error?

**Reply by The18thWarrior** [reply]
Module not found: Error: Can't resolve 'react-native' in 'C:\Users\...\dapp\node_modules\@walletconnect\universal-provider\dist'

**Reply by The18thWarrior** [reply]
Was able to get the build to work by adding the following to my package.json 

`
  "resolutions": {
    "@walletconnect/ethereum-provider": "2.7.0"
  }
`

**Reply by artschekoff** [reply]
> @walletconnect/ethereum-provider

thx! worked for me

**Reply by kenmalibu** [reply]
I tried this but couldn't get it working. Any advice?

**Reply by Buooy** [reply]
sounds like u are using npm while the rest is using yarn.

try `"overrides": { "@walletconnect/ethereum-provider": "2.7.0" }`

**Comment by The18thWarrior** [comment]
Linking associated WalletConnect Issue: https://github.com/WalletConnect/walletconnect-monorepo/issues/2313

**Reply by cryptod00c** [reply]
thanks for linking - looks like a fix just got merged - I'll test today at somepoint and report back

**Reply by The18thWarrior** [reply]
With the fix, I updated to `@web3modal/ethereum@2.4.1`, removed the `"resolutions": { "@walletconnect/ethereum-provider": "2.7.0" }` from the `package.json`, and everything worked fine

**Reply by jxom** [reply]
Latest versions of `wagmi` have the fix!

**Reply by artschekoff** [reply]
> Latest versions of `wagmi` have the fix!

after upgrade to last version, got error "lib viem not found", solved only after adding this lib...

**Reply by tmm** [reply]
viem is a peer dependency of wagmi and required to be installed.

**Comment by artschekoff** [comment]
f*ck those huge migration guides and changing basic libs ;/

https://wagmi.sh/react/migration-guide#provider--signer-terminology

omfg

Now I have to rewrite everything completely? This lib was a mistake...

**Reply by jxom** [reply]
Feel free to stay on `0.12.x` then.

**Reply by jxom** [reply]
fwiw, you do not have to "rewrite" everything, we have made the migration guide as canonical as possible so the migration is intuitive and easy to understand. Feel free to give feedback if you do not understand something there.

**Reply by cryptod00c** [reply]
migration guide is clear as day - shoutout to the team

**Reply by sofianeOuafir** [reply]
migration guide is awesome, thanks team

**Comment by Aman9723** [comment]
With `0.12.3` wallect connect is not working 

<img width="848" alt="image" src="https://github.com/wagmi-dev/wagmi/assets/65977764/93d8e588-af68-4c89-bcb1-cf899a651ce0">
